secrets/vault/vault_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limtations under the License.
+
+package vault
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/builtin/logical/transit"
+	vhttp "github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+	"gocloud.dev/secrets/driver"
+	"gocloud.dev/secrets/drivertest"
+)
+
+const (
+	keyID      = "test-secrets"
+	apiAddress = "http://127.0.0.1:0"
+)
+
+type harness struct {
+	client *api.Client
+	close  func()
+}
+
+func (h *harness) MakeDriver(ctx context.Context) (driver.Keeper, error) {
+	return &keeper{
+		keyID:  keyID,
+		client: h.client,
+	}, nil
+}
+
+func (h *harness) Close() {
+	h.close()
+}
+
+func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
+	// Start a new test server.
+	c, cleanup := testVaultServer(t)
+	// Enable the transit API.
```

Comment:
What is the transit API?

Reply:
The transit secrets engine, which allows us to call the encryption service of vault. Updated comment.